presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -250,6 +250,7 @@ predicated
 
 predicate[ParserRuleContext value]
     : comparisonOperator right=valueExpression                            #comparison
+    | comparisonOperator comparisonQuantifier '(' query ')'               #quantifiedComparison
```

Comment:
Regarding the rewrite, this is what my code is doing. In a planner it'll rewrite =ANY to IN() and <>ALL to NOT IN() which are then rewritten/optimized to use join.